UI/src/view/changeInitialViewPopup.js

```diff
@@ -0,0 +1,221 @@
+window.ChangeInitialViewPopup = function(backend, location, userRole, places, assetTypeConfig, defaultAsset, municipality) {
+
+  var options = {
+    confirmButton: 'Tallenna',
+    cancelButton: 'Peruuta',
+    successCallback: function () {
+      var selectedLocation = $('#location').val();
+      var selectedAssetType = parseInt($('#assetType').val(), 10);
+      var defaultParameters = {lon: null, lat: null, zoom: null, assetType: selectedAssetType, municipalityId: null, elyId: null};
+
+      if (selectedLocation === "currentLocation") {
+        defaultParameters.lon = location.lon;
+        defaultParameters.lat = location.lat;
+        defaultParameters.zoom = location.zoom;
+        backend.updateUserConfigurationDefaultLocation(defaultParameters);
+      }
+      else if (selectedLocation === "suomi") {
+        defaultParameters.lon = 390000;
+        defaultParameters.lat = 6900000;
+        defaultParameters.zoom = 2;
+        backend.updateUserConfigurationDefaultLocation(defaultParameters);
+      }
+      else {
+        if (userRole !== "busStopMaintainer") {
+          defaultParameters.municipalityId = parseInt(selectedLocation, 10);
+        }
+        else {
+          defaultParameters.elyId = parseInt(selectedLocation,10);
+        }
+        backend.updateUserConfigurationDefaultLocation(defaultParameters);
+      }
+
+      setTimeout(function () {
+        new GenericConfirmPopup("Laskeutumissivu päivitetty.", {type: 'alert'});
+      }, 1);
```

Comment:
This sets the timeout only at 1 millisecond

Reply:
For some reason the popup will not stay in place without the setTimeout. If it is 1 millisecond it will stay. But i will spend a little time on this to see if i can work this in other way.